js/server/modules/@arangodb/arango-collection.js

```diff
@@ -100,16 +100,24 @@ ArangoCollection.prototype.toArray = function () {
 // / @brief was docuBlock collectionTruncate
 // //////////////////////////////////////////////////////////////////////////////
 
-ArangoCollection.prototype.truncate = function () {
+ArangoCollection.prototype.truncate = function (options) {
+  if (typeof options === 'boolean') {
+    options = { waitForSync: options || false };
+  } else {
+    options = options || {};
+  }
+  if (!options.hasOwnProperty('compact')) {
+    options.compact = true;
+  }
   var cluster = require('@arangodb/cluster');
   if (cluster.isCoordinator()) {
     if (this.status() === ArangoCollection.STATUS_UNLOADED) {
       this.load();
     }
   }
-  var ret = this.TRUNCATE();
+  var ret = this.TRUNCATE(options.compact);
   // manually trigger compaction, otherwise the SST files stay large
-  if (internal.db._engine().name === "rocksdb" && this.compact) {
+  if (this.compact && options.compact) {
     this.compact();
   }
```

Comment:
This isn't needed anymore, as `TRUNCATE` does it already. And as far as I can see, the `ArangoCollection` implementation in `v8-collection.cpp` doesn't even have a `compact` method (which probably should be added, as the client-side `ArangoCollection` has it).